src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvServiceChannelProvider.java

```diff
@@ -243,7 +245,61 @@ public ServiceLocationServiceChannel findServiceLocationServiceChannel(ServiceLo
+      OrganizationId kuntaApiOrganizationId = idController.translateOrganizationId(ptvOrganizationId, KuntaApiConsts.IDENTIFIER_NAME);
+      if (kuntaApiOrganizationId == null) {
+        logger.log(Level.WARNING, () -> String.format("Failed to translate ptv organization id %s into Kunta API organization id", ptvOrganizationId));
+        return null;
+      }
+      
+      ServiceChannelApi serviceChannelApi = ptvApi.getServiceChannelApi(kuntaApiOrganizationId);
+      
+      ptvPrintableFormServiceChannelIn.setAreas(kuntaApiPtvTranslator.translateAreas(printableFormServiceChannel.getAreas()));
+      ptvPrintableFormServiceChannelIn.setAreaType(printableFormServiceChannel.getAreaType());
+      ptvPrintableFormServiceChannelIn.setAttachments(kuntaApiPtvTranslator.translateAttachments(printableFormServiceChannel.getAttachments()));
+      ptvPrintableFormServiceChannelIn.setChannelUrls(kuntaApiPtvTranslator.translateLocalizedValuesIntoLocalizedListItems(printableFormServiceChannel.getChannelUrls()));
+      ptvPrintableFormServiceChannelIn.setDeliveryAddress(kuntaApiPtvTranslator.translateDeliveryAddresses(printableFormServiceChannel.getDeliveryAddress()));
+      ptvPrintableFormServiceChannelIn.setFormIdentifier(kuntaApiPtvTranslator.translateLocalizedValuesIntoLanguageItems(printableFormServiceChannel.getFormIdentifier()));
+      ptvPrintableFormServiceChannelIn.setFormReceiver(kuntaApiPtvTranslator.translateLocalizedValuesIntoLanguageItems(printableFormServiceChannel.getFormReceiver()));
+      ptvPrintableFormServiceChannelIn.setPublishingStatus(printableFormServiceChannel.getPublishingStatus());
+      ptvPrintableFormServiceChannelIn.setServiceChannelDescriptions(kuntaApiPtvTranslator.translateLocalizedValuesIntoLocalizedListItems(printableFormServiceChannel.getDescriptions()));
+      ptvPrintableFormServiceChannelIn.setServiceChannelNames(kuntaApiPtvTranslator.translateLocalizedValuesIntoLanguageItems(printableFormServiceChannel.getNames()));
+      ptvPrintableFormServiceChannelIn.setSupportEmails(kuntaApiPtvTranslator.translateEmailsIntoLanguageItems(printableFormServiceChannel.getSupportEmails()));
+      ptvPrintableFormServiceChannelIn.setSupportPhones(kuntaApiPtvTranslator.translatePhoneNumbers(printableFormServiceChannel.getSupportPhones()));
+
+      ApiResponse<V7VmOpenApiPrintableFormChannel> response = serviceChannelApi.apiV7ServiceChannelPrintableFormByIdPut(ptvPrintableFormServiceChannelId.getId(), ptvPrintableFormServiceChannelIn);
+      if (response.isOk()) {
+        String updatedPtvChannelId = response.getResponse().getId().toString();
+        waitServiceChannelUpdate(updatedPtvChannelId);
+        return findPrintableFormChannelAfterUpdate(printableFormChannelId);
+      } else {        
+        logger.severe(() -> String.format("Failed to update service location service channel [%d] %s", response.getStatus(), response.getMessage()));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'FAILED_TO_UPDATE_SERVICE_LOCATION_SERVICE_CHANNEL' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)